File: week11/week11-4.cpp
//2000. Reverse Prefix of Word
//要把word裡,ch字母(包含)之前的自,反過來
//關鍵是先找到ch在word裡是第幾個位置
char* reversePrefix(char* word, char ch) {//字串的for迴圈
    for(int i=0;word[i]!=0;i++){//找到ch所在的位置
        if(word[i]==ch){
            for(int k=0;k<=i/2;k++){//只要迴圈做一半即可全部反轉
                int temp = word[k];//把前面反過
                word[k] = word[i-k];
                word[i-k] = temp;

            }
            break;//做完,就離開迴圈
        }

    }
    return word;
}

File: week11/week11-4.py
#2000. Reverse Prefix of Word
#要把word裡,ch字母(包含)之前的自,反過來
#關鍵是先找到ch在word裡是第幾個位置
class Solution:
    def reversePrefix(self, word: str, ch: str) -> str:
        for i in range(len(word)):#重點1:找字母
            if word[i]==ch: #找到字母了,這時候i很重要
                #重點2:把word[0]...word[i]字反過來
                return word[i::-1] + word[i+1:]#python 的特殊範圍寫法
        #若沒有提早結束,就是找不到。就原來的字回傳
        return word
